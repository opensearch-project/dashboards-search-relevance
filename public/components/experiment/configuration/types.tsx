export interface TemplateConfigurationProps {
  templateType: string;
  onBack: () => void;
  onClose: () => void;
}

export interface ConfigurationFormProps {
  templateType: string
  onSave: (formData: ConfigurationFormData) => void;
}

export interface BaseFormData {
  querySets: QuerySetOption[];
}

export interface SearchConfigFromData {
  searchConfigs: SearchConfigOption[];
}

export interface CustomizeFormData extends BaseFormData {
  calculator: string;
  scoreThreshold: string;
}

export interface UserBehaviorFormData extends BaseFormData {
  startDate: string;
  endDate: string;
  collectSignal: string;
  scoreThreshold: string;
}

export interface LLMFormData extends BaseFormData {
  modelId: string;
  scoreThreshold: string;
}

export type ConfigurationFormData =
  | CustomizeFormData
  | UserBehaviorFormData
  | LLMFormData;

export interface ConfigurationHeaderProps {
  templateType: string;
}

export interface ConfigurationActionsProps {
  onBack: () => void;
  onClose: () => void;
  onNext: () => void;
}

export interface QuerySetOption {
  label: string;
  value: string;
}

export interface SearchConfigOption {
  label: string;
  value: string;
}
