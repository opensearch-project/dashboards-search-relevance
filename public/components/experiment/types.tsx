import { CoreStart } from '../../../../../src/core/public';

export interface ExperimentPageProps {
  chrome: CoreStart['chrome'];
  application: CoreStart['application'];
  http: CoreStart['http'];
}

export interface TableContent {
  name: string;
  type: string;
  last_updated: string;
  description: string;
}

export interface ExperimentHeaderProps {
  onAddExperiment: () => void;
  history: CoreStart['history'];
}

export interface ExperimentTabsProps {
  experiments: TableContent[];
  resultListComparisonExperiments: TableContent[];
  searchConfigurations: TableContent[];
  querySets: TableContent[];
  http: CoreStart['http'];
}

export interface ExperimentTableProps {
  items: TableContent[];
}
