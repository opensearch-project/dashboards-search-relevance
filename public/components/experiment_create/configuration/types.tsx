import { RouteComponentProps } from 'react-router-dom';
import { JudgmentOption } from './form/types';

export interface TemplateConfigurationProps extends RouteComponentProps {
  templateType: string;
  onBack: () => void;
  onClose: () => void;
}

export interface ConfigurationFormProps {
  templateType: string;
  onSave: (formData: ConfigurationFormData) => void;
}

export interface BaseFormData {
  type: string;
  querySetId: string;
  size: number;
  searchConfigurationList: string[];
}

export interface SearchConfigFromData {
  searchConfigs: SearchConfigOption[];
}

export interface JudgmentOption {
  label: string;
  value: string;
}

export interface IndexOption {
  label: string;
  value: string;
}

export interface ResultListComparisonFormData extends BaseFormData {}

export interface UserBehaviorFormData extends BaseFormData {
  judgmentList: string[];
}

export interface LLMFormData extends BaseFormData {
  modelId: string;
  scoreThreshold: string;
}

export type ConfigurationFormData =
  | ResultListComparisonFormData
  | UserBehaviorFormData
  | LLMFormData;

export interface ConfigurationActionsProps {
  onBack: () => void;
  onClose: () => void;
  onNext: () => void;
}

export interface QuerySetOption {
  label: string;
  value: string;
}

export interface SearchConfigOption {
  label: string;
  value: string;
}
