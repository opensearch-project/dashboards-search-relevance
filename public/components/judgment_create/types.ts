/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export enum JudgmentType {
  LLM = 'LLM_JUDGMENT',
  UBI = 'UBI_JUDGMENT',
}

export interface ComboBoxOption {
  label: string;
  value: string;
}

export interface ModelOption extends ComboBoxOption {
  state: string;
  algorithm: string;
}

export interface JudgmentFormData {
  name: string;
  type: JudgmentType;
  // LLM specific
  querySetId?: string;
  searchConfigurationList?: string[];
  size?: number;
  modelId?: string;
  contextFields?: string[];
  tokenLimit?: number;
  ignoreFailure?: boolean;
  // UBI specific
  clickModel?: string;
  maxRank?: number;
}

export interface JudgmentCreateProps {
  http: any;
  notifications: any;
  history: any;
}
