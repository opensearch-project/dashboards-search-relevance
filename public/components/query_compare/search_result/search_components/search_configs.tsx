/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiPanel,
  EuiTitle,
  EuiSpacer,
  EuiFlexGroup,
  EuiFlexItem,
  EuiFieldText,
  EuiFormRow,
  EuiCodeEditor,
  EuiButtonIcon,
  EuiButton,
} from '@elastic/eui';
import React from 'react';
import '../../../../ace-themes/sql_console';

interface SearchConfigsPanelProps {
  isCollapsed: boolean;
  onChange: (e: React.ChangeEvent<HTMLInputElement>, setter: any) => void;
  searchIndex1: string;
  searchIndex2: string;
  setSearchIndex1: React.Dispatch<React.SetStateAction<string>>;
  setSearchIndex2: React.Dispatch<React.SetStateAction<string>>;
  queryString1: string;
  queryString2: string;
  setQueryString1: React.Dispatch<React.SetStateAction<string>>;
  setQueryString2: React.Dispatch<React.SetStateAction<string>>;
  setIsCollapsed: React.Dispatch<React.SetStateAction<boolean>>;
}

export const SearchConfigsPanel = ({
  isCollapsed,
  onChange,
  searchIndex1,
  searchIndex2,
  setSearchIndex1,
  setSearchIndex2,
  queryString1,
  queryString2,
  setQueryString1,
  setQueryString2,
  setIsCollapsed,
}: SearchConfigsPanelProps) => {
  return (
    <EuiPanel paddingSize="m">
      <EuiFlexGroup>
        <EuiFlexItem grow={false}>
          <EuiButtonIcon
            color="text"
            display="base"
            iconType={isCollapsed ? 'arrowDown' : 'arrowRight'}
            iconSize="m"
            size="s"
            aria-label="Next"
            onClick={() => setIsCollapsed(!isCollapsed)}
          />
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiTitle size="s">
            <h2>Search Configs</h2>
          </EuiTitle>
        </EuiFlexItem>
      </EuiFlexGroup>
      {!isCollapsed && (
        <>
          <EuiSpacer size="l" />
          <EuiFlexGroup>
            <EuiFlexItem>
              <div>
                <EuiFormRow
                  fullWidth
                  label="Search Index 1"
                  helpText="Enter the index or index pattern for the query"
                >
                  <EuiFieldText
                    name="searchIndex1"
                    value={searchIndex1}
                    onChange={(e) => setSearchIndex1(e.target.value)}
                    aria-label="searchIndex1"
                  />
                </EuiFormRow>

                <EuiFormRow
                  fullWidth
                  label="Query String 1"
                  helpText="Enter the DSL query for searching over the index"
                >
                  <EuiCodeEditor
                    mode="sql"
                    theme="sql_console"
                    width="100%"
                    height="10rem"
                    value={queryString1}
                    onChange={setQueryString1}
                    showPrintMargin={false}
                    setOptions={{
                      fontSize: '14px',
                      enableBasicAutocompletion: true,
                      enableLiveAutocompletion: true,
                    }}
                    aria-label="Code Editor"
                  />
                </EuiFormRow>
                <EuiButton
                  onClick={() => {
                    setSearchIndex1('');
                    setQueryString1('');
                  }}
                >
                  Clear
                </EuiButton>
              </div>
            </EuiFlexItem>
            <EuiFlexItem>
              <div>
                <EuiFormRow
                  label="Search Index 2"
                  helpText="Enter the index or index pattern for the query"
                >
                  <EuiFieldText
                    name="searchIndex2"
                    value={searchIndex2}
                    onChange={(e) => setSearchIndex2(e.target.value)}
                    aria-label="searchIndex2"
                  />
                </EuiFormRow>

                <EuiFormRow
                  fullWidth
                  label="Query String 2"
                  helpText="Enter the DSL query for searching over the index"
                >
                  <EuiCodeEditor
                    mode="sql"
                    theme="sql_console"
                    width="100%"
                    height="10rem"
                    value={queryString2}
                    onChange={setQueryString2}
                    showPrintMargin={false}
                    setOptions={{
                      fontSize: '14px',
                      enableBasicAutocompletion: true,
                      enableLiveAutocompletion: true,
                    }}
                    aria-label="Code Editor"
                  />
                </EuiFormRow>
                <EuiButton
                  onClick={() => {
                    setSearchIndex2('');
                    setQueryString2('');
                  }}
                >
                  Clear
                </EuiButton>
              </div>
            </EuiFlexItem>
          </EuiFlexGroup>
        </>
      )}
    </EuiPanel>
  );
};
