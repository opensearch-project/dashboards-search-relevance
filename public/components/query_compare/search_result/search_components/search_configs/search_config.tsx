/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { FunctionComponent } from 'react';
import {
  EuiTitle,
  EuiSpacer,
  EuiFormRow,
  EuiSelect,
  EuiCodeEditor,
  EuiText,
  EuiButtonEmpty,
} from '@elastic/eui';

import { useSearchRelevanceContext } from '../../../../../contexts';
import { QueryError, QueryStringError } from '../../../../../types/index';

interface SearchConfigProps {
  queryNumber: 1 | 2;
  queryString: string;
  setQueryString: React.Dispatch<React.SetStateAction<string>>;
  selectedIndex: string;
  setSelectedIndex: React.Dispatch<React.SetStateAction<string>>;
  queryError: QueryError;
  setQueryError: React.Dispatch<React.SetStateAction<QueryError>>;
}

export const SearchConfig: FunctionComponent<SearchConfigProps> = ({
  queryNumber,
  queryString,
  setQueryString,
  selectedIndex,
  setSelectedIndex,
  queryError,
  setQueryError,
}) => {
  const { documentsIndexes, setShowFlyout } = useSearchRelevanceContext();

  // On select index
  const onChangeSelectedIndex: React.ChangeEventHandler<HTMLSelectElement> = (e) => {
    setSelectedIndex(e.target.value);

    setQueryError({
      ...queryError,
      selectIndex: '',
    });
  };

  // Select index on blur
  const selectIndexOnBlur = () => {
    // If Index Select on blur without selecting an index, show error
    if (!selectedIndex.length) {
      setQueryError({
        ...queryError,
        selectIndex: 'An index is required. Select an index.',
      });
    }
  };

  // On change query string
  const onChangeQueryString = (value: string) => {
    setQueryString(value);
    setQueryError({
      ...queryError,
      queryString: '',
    });
  };

  // Code editor on blur
  const codeEditorOnBlur = () => {
    // If no query string on blur, show error
    if (!queryString.length) {
      setQueryError({
        ...queryError,
        queryString: QueryStringError.empty,
      });
    }
  };

  return (
    <>
      <EuiTitle size="xs">
        <h2 style={{ fontWeight: '300', fontSize: '21px' }}>Query {queryNumber}</h2>
      </EuiTitle>
      <EuiSpacer size="m" />
      <EuiFormRow
        fullWidth
        label="Index"
        error={!!queryError.selectIndex.length && <span>{queryError.selectIndex}</span>}
        isInvalid={!!queryError.selectIndex.length}
      >
        <EuiSelect
          hasNoInitialSelection={true}
          options={documentsIndexes.map(({ index }) => ({
            value: index,
            text: index,
          }))}
          aria-label="Search Index"
          onChange={onChangeSelectedIndex}
          value={selectedIndex}
          onBlur={selectIndexOnBlur}
        />
      </EuiFormRow>
      <EuiFormRow
        fullWidth
        label="Query"
        error={!!queryError.queryString.length && <span>{queryError.queryString}</span>}
        isInvalid={!!queryError.queryString.length}
        labelAppend={
          <EuiText size="xs">
            <EuiButtonEmpty size="xs" color="primary" onClick={() => setShowFlyout(true)}>
              Help
            </EuiButtonEmpty>
          </EuiText>
        }
      >
        <EuiCodeEditor
          mode="json"
          theme="sql_console"
          width="100%"
          height="10rem"
          value={queryString}
          onChange={onChangeQueryString}
          showPrintMargin={false}
          setOptions={{
            fontSize: '14px',
            enableBasicAutocompletion: true,
            enableLiveAutocompletion: true,
          }}
          aria-label="Code Editor"
          onBlur={codeEditorOnBlur}
          tabSize={2}
        />
      </EuiFormRow>
      <EuiText>
        <p style={{ fontSize: '14px', fontWeight: '400', lineHeight: '18px' }}>
          Enter a query in OpenSearch Query DSL. Use %SearchText% to refer to the text in the search
          bar.
        </p>
      </EuiText>
    </>
  );
};
