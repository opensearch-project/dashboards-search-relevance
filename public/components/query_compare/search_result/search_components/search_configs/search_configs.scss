/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.search-relevance-config:nth-child(1) {
  margin: 0;
  padding: 12px;
  border-right: 1px solid $euiBorderColor;
}

.left-right-borders {
  border-left: 0;
  border-right: 0;
}