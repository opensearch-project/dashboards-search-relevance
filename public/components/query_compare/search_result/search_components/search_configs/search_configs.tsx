/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiFlexGroup, EuiFlexItem, EuiPanel } from '@elastic/eui';
import React from 'react';

import '../../../../../ace-themes/sql_console';
import { useSearchRelevanceContext } from '../../../../../contexts';
import { SearchConfig } from './search_config';

import { CoreStart, MountPoint } from '../../../../../../../../src/core/public';
import { DataSourceManagementPluginSetup } from '../../../../../../../../src/plugins/data_source_management/public';
import { DataSourceOption } from '../../../../../../../../src/plugins/data_source_management/public/components/data_source_selector/data_source_selector';
import { NavigationPublicPluginStart } from '../../../../../../../../src/plugins/navigation/public';
import { QueryError } from '../../../../../../public/types/index';
import './search_configs.scss';

interface SearchConfigsPanelProps {
  queryString1: string;
  queryString2: string;
  setQueryString1: React.Dispatch<React.SetStateAction<string>>;
  setQueryString2: React.Dispatch<React.SetStateAction<string>>;
  queryError1: QueryError;
  queryError2: QueryError;
  setQueryError1: React.Dispatch<React.SetStateAction<QueryError>>;
  setQueryError2: React.Dispatch<React.SetStateAction<QueryError>>;
  dataSourceEnabled: boolean;
  savedObjects: CoreStart['savedObjects'];
  notifications: CoreStart['notifications'];
  dataSourceManagement: DataSourceManagementPluginSetup;
  navigation: NavigationPublicPluginStart;
  setActionMenu: (menuMount: MountPoint | undefined) => void;
  dataSourceOptions: DataSourceOption[];
}

export const SearchConfigsPanel = ({
  queryString1,
  queryString2,
  setQueryString1,
  setQueryString2,
  queryError1,
  queryError2,
  setQueryError1,
  setQueryError2,
  dataSourceEnabled,
  savedObjects,
  dataSourceManagement,
  setActionMenu,
  navigation,
  dataSourceOptions,
  notifications
}: SearchConfigsPanelProps) => {
  const {
    selectedIndex1,
    setSelectedIndex1,
    selectedIndex2,
    setSelectedIndex2,
    pipeline1,
    setPipeline1,
    pipeline2,
    setPipeline2,
    datasource1,
    setDataSource1,
    datasource2,
    setDataSource2,
  } = useSearchRelevanceContext();

  return (
    <EuiPanel
      hasShadow={false}
      color="transparent"
      grow={false}
      borderRadius="none"
      className="left-right-borders"
    >
      <EuiFlexGroup>
        <EuiFlexItem className="search-relevance-config">
          <SearchConfig
            queryNumber={1}
            queryString={queryString1}
            setQueryString={setQueryString1}
            selectedIndex={selectedIndex1}
            setSelectedIndex={setSelectedIndex1}
            queryError={queryError1}
            setQueryError={setQueryError1}
            pipeline={pipeline1}
            setPipeline={setPipeline1}
            setDataSource={setDataSource1}
            dataSourceEnabled={dataSourceEnabled}
            savedObjects={savedObjects}
            dataSourceManagement={dataSourceManagement}
            navigation={navigation}
            setActionMenu={setActionMenu}
            dataSourceOptions={dataSourceOptions}
            notifications={notifications}
          />
        </EuiFlexItem>
        <EuiFlexItem className="search-relevance-config">
          <SearchConfig
            queryNumber={2}
            queryString={queryString2}
            setQueryString={setQueryString2}
            selectedIndex={selectedIndex2}
            setSelectedIndex={setSelectedIndex2}
            queryError={queryError2}
            setQueryError={setQueryError2}
            pipeline={pipeline2}
            setPipeline={setPipeline2}
            setDataSource={setDataSource2}
            dataSourceEnabled={dataSourceEnabled}
            savedObjects={savedObjects}
            dataSourceManagement={dataSourceManagement}
            navigation={navigation}
            setActionMenu={setActionMenu}
            dataSourceOptions={dataSourceOptions}
            notifications={notifications}
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiPanel>
  );
};
