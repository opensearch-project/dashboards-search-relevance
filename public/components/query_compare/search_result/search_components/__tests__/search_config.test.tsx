/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { SearchConfig } from '../search_configs/search_config';
import { SearchRelevanceContextProvider } from '../../../../../contexts';
import { TEST_QUERY_STRING } from '../../../../../../test/constants';
import { initialQueryErrorState } from '../../../../../../public/types/index';

describe('Flyout component', () => {
  configure({ adapter: new Adapter() });

  it('Renders flyout component', async () => {
    const setQueryString = jest.fn();
    const setSelectedIndex = jest.fn();
    const setQueryError = jest.fn();
    const wrapper = mount(
      <SearchRelevanceContextProvider>
        <SearchConfig
          queryNumber={1}
          queryString={TEST_QUERY_STRING}
          setQueryString={setQueryString}
          selectedIndex={''}
          setSelectedIndex={setSelectedIndex}
          queryError={initialQueryErrorState}
          setQueryError={setQueryError}
        />
      </SearchRelevanceContextProvider>
    );

    wrapper.update();
    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
      wrapper.find('EuiCodeEditor').prop('onChange')?.({ target: { value: '' } });
      wrapper.find('EuiSelect').prop('onChange')?.({ target: {} });
      wrapper.find('EuiSelect').prop('onBlur')?.({ target: {} });
    });
    expect(setQueryString).toHaveBeenCalledTimes(1);
    expect(setSelectedIndex).toHaveBeenCalledTimes(1);
    expect(setQueryError).toHaveBeenCalledTimes(3);
  });
});
