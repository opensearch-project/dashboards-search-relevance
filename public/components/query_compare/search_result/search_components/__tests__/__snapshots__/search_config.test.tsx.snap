// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flyout component Renders flyout component 1`] = `
<SearchRelevanceContextProvider>
  <SearchConfig
    queryError={
      Object {
        "errorResponse": Object {
          "body": "",
          "statusCode": 200,
        },
        "queryString": "",
        "selectIndex": "",
      }
    }
    queryNumber={1}
    queryString="{}"
    selectedIndex=""
    setQueryError={[MockFunction]}
    setQueryString={[MockFunction]}
    setSelectedIndex={[MockFunction]}
  >
    <EuiTitle
      size="xs"
    >
      <h2
        className="euiTitle euiTitle--xsmall"
        style={
          Object {
            "fontSize": "21px",
            "fontWeight": "300",
          }
        }
      >
        Query 
        1
      </h2>
    </EuiTitle>
    <EuiSpacer
      size="m"
    >
      <div
        className="euiSpacer euiSpacer--m"
      />
    </EuiSpacer>
    <EuiFormRow
      describedByIds={Array []}
      display="row"
      error={false}
      fullWidth={true}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      isInvalid={false}
      label="Index"
      labelType="label"
    >
      <div
        className="euiFormRow euiFormRow--fullWidth"
        id="random_html_id-row"
      >
        <div
          className="euiFormRow__labelWrapper"
        >
          <EuiFormLabel
            aria-invalid={false}
            className="euiFormRow__label"
            htmlFor="random_html_id"
            isFocused={false}
            isInvalid={false}
            type="label"
          >
            <label
              aria-invalid={false}
              className="euiFormLabel euiFormRow__label"
              htmlFor="random_html_id"
            >
              Index
            </label>
          </EuiFormLabel>
        </div>
        <div
          className="euiFormRow__fieldWrapper"
        >
          <EuiSelect
            aria-label="Search Index"
            hasNoInitialSelection={true}
            id="random_html_id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            options={Array []}
            value=""
          >
            <EuiFormControlLayout
              compressed={false}
              fullWidth={false}
              icon={
                Object {
                  "side": "right",
                  "type": "arrowDown",
                }
              }
              inputId="random_html_id"
              isLoading={false}
            >
              <div
                className="euiFormControlLayout"
              >
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <EuiValidatableControl>
                    <select
                      aria-label="Search Index"
                      className="euiSelect"
                      id="random_html_id"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onMouseUp={[Function]}
                      value=""
                    >
                      <option
                        disabled={true}
                        hidden={true}
                        style={
                          Object {
                            "display": "none",
                          }
                        }
                        value=""
                      >
                        Â 
                      </option>
                    </select>
                  </EuiValidatableControl>
                  <EuiFormControlLayoutIcons
                    compressed={false}
                    icon={
                      Object {
                        "side": "right",
                        "type": "arrowDown",
                      }
                    }
                    isLoading={false}
                  >
                    <div
                      className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                    >
                      <EuiFormControlLayoutCustomIcon
                        size="m"
                        type="arrowDown"
                      >
                        <span
                          className="euiFormControlLayoutCustomIcon"
                        >
                          <EuiIcon
                            aria-hidden="true"
                            className="euiFormControlLayoutCustomIcon__icon"
                            size="m"
                            type="arrowDown"
                          >
                            <EuiIconEmpty
                              aria-hidden={true}
                              className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                              focusable="false"
                              role="img"
                              style={null}
                            >
                              <svg
                                aria-hidden={true}
                                className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                                focusable="false"
                                height={16}
                                role="img"
                                style={null}
                                viewBox="0 0 16 16"
                                width={16}
                                xmlns="http://www.w3.org/2000/svg"
                              />
                            </EuiIconEmpty>
                          </EuiIcon>
                        </span>
                      </EuiFormControlLayoutCustomIcon>
                    </div>
                  </EuiFormControlLayoutIcons>
                </div>
              </div>
            </EuiFormControlLayout>
          </EuiSelect>
        </div>
      </div>
    </EuiFormRow>
    <EuiFormRow
      describedByIds={Array []}
      display="row"
      error={false}
      fullWidth={true}
      hasChildLabel={true}
      hasEmptyLabelSpace={false}
      helpText={
        <p>
          Enter a query in
           
          <a
            href="https://opensearch.org/docs/latest/query-dsl/index/"
          >
            OpenSearch Query DSL
          </a>
          . Use %SearchText% to refer to the text in the search bar
        </p>
      }
      isInvalid={false}
      label="Query"
      labelAppend={
        <EuiText
          size="xs"
        >
          <EuiButtonEmpty
            color="primary"
            onClick={[Function]}
            size="xs"
          >
            Help
          </EuiButtonEmpty>
        </EuiText>
      }
      labelType="label"
    >
      <div
        className="euiFormRow euiFormRow--fullWidth"
        id="random_html_id-row"
      >
        <div
          className="euiFormRow__labelWrapper"
        >
          <EuiFormLabel
            aria-invalid={false}
            className="euiFormRow__label"
            htmlFor="random_html_id"
            isFocused={false}
            isInvalid={false}
            type="label"
          >
            <label
              aria-invalid={false}
              className="euiFormLabel euiFormRow__label"
              htmlFor="random_html_id"
            >
              Query
            </label>
          </EuiFormLabel>
           
          <EuiText
            size="xs"
          >
            <div
              className="euiText euiText--extraSmall"
            >
              <EuiButtonEmpty
                color="primary"
                onClick={[Function]}
                size="xs"
              >
                <button
                  className="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <EuiButtonContent
                    className="euiButtonEmpty__content"
                    iconSide="left"
                    iconSize="s"
                    textProps={
                      Object {
                        "className": "euiButtonEmpty__text",
                      }
                    }
                  >
                    <span
                      className="euiButtonContent euiButtonEmpty__content"
                    >
                      <span
                        className="euiButtonEmpty__text"
                      >
                        Help
                      </span>
                    </span>
                  </EuiButtonContent>
                </button>
              </EuiButtonEmpty>
            </div>
          </EuiText>
        </div>
        <div
          className="euiFormRow__fieldWrapper"
        >
          <EuiCodeEditor
            aria-describedby="random_html_id-help-0"
            aria-label="Code Editor"
            height="10rem"
            id="random_html_id"
            mode="json"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            setOptions={
              Object {
                "enableBasicAutocompletion": true,
                "enableLiveAutocompletion": true,
                "fontSize": "14px",
              }
            }
            showPrintMargin={false}
            tabSize={2}
            theme="sql_console"
            value="{}"
            width="100%"
          >
            <div
              className="euiCodeEditorWrapper"
              data-test-subj="codeEditorContainer"
              style={
                Object {
                  "height": "10rem",
                  "width": "100%",
                }
              }
            >
              <button
                className="euiCodeEditorKeyboardHint"
                data-test-subj="codeEditorHint"
                id="random_html_id"
                onClick={[Function]}
              >
                <p
                  className="euiText"
                >
                  <EuiI18n
                    default="Press Enter to start editing."
                    token="euiCodeEditor.startEditing"
                  >
                    Press Enter to start editing.
                  </EuiI18n>
                </p>
                <p
                  className="euiText"
                >
                  <EuiI18n
                    default="When you're done, press Escape to stop editing."
                    token="euiCodeEditor.stopEditing"
                  >
                    When you're done, press Escape to stop editing.
                  </EuiI18n>
                </p>
              </button>
              <ReactAce
                aria-describedby="random_html_id-help-0"
                aria-label="Code Editor"
                commands={
                  Array [
                    Object {
                      "bindKey": Object {
                        "mac": "Esc",
                        "win": "Esc",
                      },
                      "exec": [Function],
                      "name": "stopEditingOnEsc",
                    },
                  ]
                }
                cursorStart={1}
                editorProps={
                  Object {
                    "$blockScrolling": Infinity,
                  }
                }
                enableBasicAutocompletion={false}
                enableLiveAutocompletion={false}
                enableSnippets={false}
                focus={false}
                fontSize={12}
                height="10rem"
                highlightActiveLine={true}
                id="random_html_id"
                maxLines={null}
                minLines={null}
                mode="json"
                name="random_html_id"
                navigateToFileEnd={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onLoad={null}
                onPaste={null}
                onScroll={null}
                placeholder={null}
                readOnly={false}
                scrollMargin={
                  Array [
                    0,
                    0,
                    0,
                    0,
                  ]
                }
                setOptions={
                  Object {
                    "enableBasicAutocompletion": true,
                    "enableLiveAutocompletion": true,
                    "fontSize": "14px",
                  }
                }
                showGutter={true}
                showPrintMargin={false}
                style={Object {}}
                tabSize={2}
                theme="sql_console"
                value="{}"
                width="100%"
                wrapEnabled={false}
              >
                <div
                  id="random_html_id"
                  style={
                    Object {
                      "height": "10rem",
                      "width": "100%",
                    }
                  }
                />
              </ReactAce>
            </div>
          </EuiCodeEditor>
          <EuiFormHelpText
            className="euiFormRow__text"
            id="random_html_id-help-0"
            key="0"
          >
            <div
              className="euiFormHelpText euiFormRow__text"
              id="random_html_id-help-0"
            >
              <p>
                Enter a query in
                 
                <a
                  href="https://opensearch.org/docs/latest/query-dsl/index/"
                >
                  OpenSearch Query DSL
                </a>
                . Use %SearchText% to refer to the text in the search bar
              </p>
            </div>
          </EuiFormHelpText>
        </div>
      </div>
    </EuiFormRow>
  </SearchConfig>
</SearchRelevanceContextProvider>
`;
